src/main/java/com/dubture/jenkins/digitalocean/SlaveTemplate.java

```diff
@@ -248,13 +248,33 @@ public ListBoxModel doFillSizeIdItems(@RelativePath("..") @QueryParameter String
 
         public ListBoxModel doFillImageIdItems(@RelativePath("..") @QueryParameter String authToken) throws Exception {
 
-            SortedMap<String, Image> availableSizes = DigitalOcean.getAvailableImages(authToken);
+            SortedMap<String, Image> availableImages = DigitalOcean.getAvailableImages(authToken);
             ListBoxModel model = new ListBoxModel();
 
-            for (Map.Entry<String, Image> entry : availableSizes.entrySet()) {
-                // Reference image IDs instead of slugs so that we can build
-                // images based upon snapshots as well as standard images
-                model.add(entry.getKey(), entry.getValue().getId().toString());
+            for (Map.Entry<String, Image> entry : availableImages.entrySet()) {
+                final Image image = entry.getValue();
+                final String value;
+
+                switch (image.getType()) {
+                    case SNAPSHOT:
+                        value = image.getSlug();
+                        break;
+
+                    case BACKUP:
+                        // Use the image ID instead of the slug (which isn't available anyway)
+                        // so that we can build images based upon snapshots as well as standard images
+                        value = image.getId().toString();
+                        break;
+
+                    default:
+                        LOGGER.log(Level.WARNING, "Ignoring image of type " + image.getType());
+                        value = null;
```

Comment:
are you absolutely sure about this? the previous call would call the getId method even in this case!